Adds updated deltasplit75

itsaferbie opened the pull request:
This pull request is for fixing the naming convention of the deltasplit75 project, bringing it up to the other keyboards found in qmk. I didn't fix the issue with the protosplit keymap throwing an error when just compiling with v2. As I don't own a protosplit. I also included my own keymap.

itsaferbie commented:
For some reason pull request created on main repo instead of creators repo.

jackhumbert commented:
We'd be happy to accept this here too! I think we were just waiting on these changes in #1358 .

itsaferbie commented:
I have submitted the request to the creators repo, so hopefully they will merge with mine which would include the structure change, and they can fix the compilation error. Then we can have this included in qmk!

itsaferbie commented:
Since the creator is away for military I am deciding to open my Pull request again, I have renamed the folders and files to match that of qmk's other keyboards, and I have added to blank spaces to the end of the protosplit keymap to for it to make correctly with v2.

itsaferbie commented:
If you notice anything that might not be correct, just let me know and I'll try my best to fix it.

jackhumbert commented on keyboards/deltasplit75/pro_micro.h:
```diff
@@ -0,0 +1,362 @@
+/*
```
This file shouldn't be needed anymore - we have `pro_micro.h` in the `/drivers/avr` folder now ;)

jackhumbert commented on keyboards/deltasplit75/protosplit/protosplit.c:
```diff
@@ -0,0 +1,32 @@
+
+    #ifdef AUDIO_ENABLE
+        _delay_ms(20); // gets rid of tick
+        PLAY_NOTE_ARRAY(tone_startup, false, 0);
```
The startup song is actually played by default - you don't need to have any of this in here :)

jackhumbert commented on keyboards/deltasplit75/protosplit/protosplit.c:
```diff
@@ -0,0 +1,32 @@
+
+void shutdown_user(void) {
+    #ifdef AUDIO_ENABLE
+        PLAY_NOTE_ARRAY(tone_goodbye, false, 0);
```
Same goes for this :)

jackhumbert commented on keyboards/deltasplit75/v2/v2.h:
```diff
@@ -0,0 +1,37 @@
+#ifndef V2_H
+#define V2_H
+
+#include "../deltasplit75.h"
```
The keyboard folder should be in the path, so I think this can just be `deltasplit75.h` - does that not work?

jackhumbert commented on keyboards/deltasplit75/config.h:
```diff
@@ -0,0 +1,29 @@
+
+#include "config_common.h"
+
+#ifdef SUBPROJECT_RightB
```
This should be `protosplit`, right? I'm not seeing `RightB` anywhere else in here.

jackhumbert commented:
Cool! Can you remove the .pdf as well? We're trying to move files like that over to https://github.com/qmk/qmk.fm :) (feel free to make a pull request there - there's more info on that [here](https://docs.qmk.fm/adding_a_keyboard_to_qmk.html#imagehardware-files))

itsaferbie commented:
Alright, I have gone through and updated the things that you suggested. And will also look into making a pull request towards qmk.fm for the build guide.

jackhumbert commented on keyboards/deltasplit75/protosplit/protosplit.c:
```diff
@@ -1,32 +1,19 @@
+#include "deltasplit75.h"
+
+#ifdef AUDIO_ENABLE
+    float tone_startup[][2] = SONG(STARTUP_SOUND);
```
This section won't be needed either then! :)

jackhumbert commented:
It's still [failing on Travis](https://travis-ci.org/qmk/qmk_firmware/builds/258300413) - it looks like there's a mismatch between the keymap's KEYMAP and what the header file defines.

jackhumbert commented on keyboards/deltasplit75/v2/v2.h:
```diff
@@ -0,0 +1,37 @@
+//void promicro_bootloader_jmp(bool program);
+//matrix is defined in a weird way here; the layout on both sides are asymmetrical, but the "matrix" is symmetrical but with empty gaps
+//the last column is defined as a separate row because the firmware currently doesnt support more than 8 columns (this layout has 9 columns per side) K45 and K110 are the Bs on both sides; K53 and K106 are extra keys for ISO
+#define KEYMAP( \
```
One way to fix the error would be to define two KEYMAPs, `KEYMAP_V2` and `KEYMAP_PROTOSPLIT` with different numbers of arguments (88 and 90 it looks like), and use those in the keymaps. 

The other (possibly easier) way would be to just accept 90 in both, and always have `KC_NO` in the keymaps that don't use them.

itsaferbie replied:
Wow, that's really cool. I'll try that.

jackhumbert commented:
If the KEYMAP is the only difference between the revisions, you could get rid of one of them!

itsaferbie commented:
Good to know! I went ahead and checked the configs of both and they are the same, so removed the protosplit directory.

jackhumbert commented:
Awesome :) thanks for making all of these changes!
